NGitLab.Mock/Clients/MilestoneClient.cs

```diff
@@ -26,35 +25,46 @@ public MilestoneClient(ClientContext context, int id, MilestoneScope scope)
             {
                 using (Context.BeginOperationScope())
                 {
-                    var project = GetProject(_resourceId, ProjectPermission.View);
-                    return FindMilestone(id, project)?.ToClientMilestone();
+                    return GetMilestone(id, false).ToClientMilestone();
                 }
             }
         }
 
         public IEnumerable<Models.Milestone> All => Get(new MilestoneQuery());
 
-        public MilestoneScope Scope => throw new NotImplementedException();
+        public MilestoneScope Scope { get; }
 
         public Models.Milestone Activate(int milestoneId)
         {
             using (Context.BeginOperationScope())
             {
-                var milestone = new Milestone();
+                var milestone = GetMilestone(milestoneId, true);
+                milestone.State = MilestoneState.active;
+                return milestone.ToClientMilestone();
+            }
+        }
 
-                if (_scope == MilestoneScope.Groups)
-                {
-                    var group = GetGroup(_resourceId, GroupPermission.Edit);
-                    milestone = FindMilestone(milestoneId, group) ?? throw new GitLabNotFoundException($"Cannot find milestone with ID {milestoneId}");
-                }
-                else if (_scope == MilestoneScope.Projects)
+        public IEnumerable<Models.MergeRequest> GetMergeRequests(int milestoneId)
+        {
+            using (Context.BeginOperationScope())
+            {
+                var milestone = GetMilestone(milestoneId, false);
+                IEnumerable<MergeRequest> mergeRequests;
+
+                switch (Scope)
                 {
-                    var project = GetProject(_resourceId, ProjectPermission.Edit);
-                    milestone = FindMilestone(milestoneId, project) ?? throw new GitLabNotFoundException($"Cannot find milestone with ID {milestoneId}");
+                    case MilestoneScope.Groups:
+                        mergeRequests = milestone.Group.MergeRequests;
+                        break;
+                    case MilestoneScope.Projects:
+                        mergeRequests = milestone.Project.MergeRequests;
+                        break;
+                    default:
+                        throw new NotImplementedException();
```

Comment:
Can you add a message in the exception? As it's not totally not implemented :)

(Same for the other methods that have NotImplementedException() with some implementation